### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by duskotmane**
# Features ⭐ ⭐ ⭐ 

- [ ] - Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [ ] Install and setup devise
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [ ] Modify the views in devise for registration and login to match the wireframes from the Sneak Peek

- I am working solo in this project by the way
 
solo

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Hi @otmaneechchafyky , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [ ] Kindly note that this is a pair programming activity. Your partner should contribute to the repo 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by duskotmane** [comment]
@frank1738 please check the PR description, I mention that I am working solo and I have no partners this week

**Comment by ghost** [comment]
> @frank1738 please check the PR description, I mention that I am working solo and I have no partners this week

Hello @otmaneechchafyky . Thanks for the update. Looks like you updated the information after a review was made. Kindly resubmit your PR for another review

**Comment by duskotmane** [comment]
@frank1738 I did brother, if you have time can you review it please.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,3 +7,18 @@ def current_user
...
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
+  end
+
+  def configure_devise_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |user|
+      user.permit(:name, :email, :password, :password_confirmation)
+    end
+  end
+end
```
Comment (mckent05) [inline app/controllers/application_controller.rb]: - You have done well so far, but you have created a base controller inside a base controller, I suggest that you have just one base controller and ensure your application works correctly, please fix this 👍 👍 
- Also in the parent controller, according to the project requirements, you are required to remove the `current_user` method as this is already handled by `devise`. Please fix this 😄 
- Please also ensure that you authenticate all users before any action is accessed, please fix this 😄

**Comment by ghost** [comment]
> @frank1738 I did brother, if you have time can you review it please.

A review will be made shortly👍

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (mckent05) [inline app/models/user.rb]: - Good job done so far, but you have repeated code in your application, I suggest you have just one devise configuration and ensure it is according to the project requirements, please fix this 😄

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @otmaneechchafyky, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding devise to your application 👏 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (duskotmane) [reply app/controllers/application_controller.rb]: done

File: app/models/user.rb (hunk 2 above)
Comment (duskotmane) [reply app/models/user.rb]: done

**Review by nedjwab — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi  @otmaneechchafyky                    👋

Your project is complete! ✔️✔  🏁

Your code was very professional 💯

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉

 
## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors


- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by duskotmane** [comment]
Thank you

**Pull request merged**